### PR TITLE
[ Feat ] 반례 추가 모달 구현

### PR DESCRIPTION
**Pull request by j-nary**
## ✅ Done Task
  - [x] 반례 추가 모달 구현
  - [x] API 연결

## 📸 Screenshot

https://github.com/user-attachments/assets/0ea8206c-5869-478e-a8a1-46145e809d08

### CODE REVIEW

File: apps/client/src/app/[user]/edge-case/components/EdgeCaseController/EdgeCaseCreateForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      input: "",
+      output: "",
+    },
+  });
+  const { mutate: createEdgeCase } = useEdgeCaseMutation();
+
+  const handleSubmit = (values: z.infer<typeof edgeCaseCreateFormSchema>) => {
+    const { problem, input, output } = values;
+
+    const link = isNaN(Number(problem))
```
Comment (ptyoiy) [inline apps/client/src/app/[user]/edge-case/components/EdgeCaseController/EdgeCaseCreateForm/index.tsx]: `isNaN`의 인자에 Number 형 변환을 하셨는데, `isNaN` 자체가 이미 그 동작을 합니다.

무슨 말이냐 하면, 일단 여기서 사용하신 `isNaN`은 `window`객체의 메서드고, `Number.isNaN`은 `Number`객체의 메서드에요. 둘이 하는 일은 똑같은데 동작이 달라서 보통은 동작 예측의 일관성을 위해 후자의 것을 사용하는 게 좋아요. 
(하지만 이 부분은 그 보통의 상황이 아니라서 Number 형 변환만 빼주면 되긴 합니다.)
1. `isNaN`
**인자로 받은 값을 먼저 숫자(Number) 타입으로 강제 변환**하려고 시도합니다. 그 변환된 결과가 NaN이면 true를 반환합니다.
_`isNaN("hello"); // true`_
2. `Number.isNaN`
오직 인자의 타입이 `Number`이면서 **그 값이 정확히 `NaN`일 경우에만 `true`를 반환**합니다.
_`Number.isNaN("hello") // false`_

근데 그렇다고 `isNaN`을 쓰면 biome이 뭐라 하니까 대신 사용할 만한 건 이렇게 2개가 되겠네요.
1. `/^\d+$/.test(problem)`
2. Number.isNaN(+problem)

Comment (ptyoiy) [reply apps/client/src/app/[user]/edge-case/components/EdgeCaseController/EdgeCaseCreateForm/index.tsx]: 추가적으로 폼 입력 때 입력 값을 통제해줄 수 있도록 `edgeCaseCreateFormSchema`도 수정해서 문제 번호 혹은 링크만 받는 게 좋겠어요. 
```ts
export const edgeCaseCreateFormSchema = z.object({
  // 숫자로만 된 string 혹은 url형식인 string
  problem: z.union([z.string().regex(/^\d+$/), z.string().url()], {
    // 공통 에러 메세지
    errorMap: () => ({ message: "문제 번호 또는 유효한 URL을 입력해주세요." }),
  }),
  ...
});
```

File: apps/client/src/app/[user]/edge-case/page.tsx
Comment (ptyoiy) [inline apps/client/src/app/[user]/edge-case/page.tsx:1]: 클라 컴포넌트로 바꾸셨는데, 충분히 서버 컴포넌트로 할 수 있다고 생각해요. `WrongSolvedList` 컴포넌트의 내용을 보니 계속 서버 컴포넌트로 사용할 수 있을 것 같아서 클라 page로 하기엔 아까워요.
아래의 EdgeCase 컴포넌트 2개는 하나의 use client 컴포넌트로 묶고 hydration 해주면 되겠어요.

**Review by ptyoiy — APPROVED** [review]
소소한 코멘트가 있는데 그것만 확인해주심 되겠습니다. LGTM!

**Review by rtttr1 — APPROVED** [review]
고생하셨습니다~~!!

File: apps/client/src/app/[user]/edge-case/components/EdgeCaseController/EdgeCaseCreateForm/index.tsx (hunk 1 above)
Comment (j-nary) [reply apps/client/src/app/[user]/edge-case/components/EdgeCaseController/EdgeCaseCreateForm/index.tsx]: 꼼꼼한 피드백 감사합니닷! 🙇‍♀️
